src/Directory.Build.props

```diff
@@ -1,3 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) .NET Foundation and contributors. All rights reserved. Licensed under the MIT license. See License.txt in the project root for full license information. -->
 <Project>
-     <Import Project="$([MSBuild]::GetPathOfFileAbove('Common.props'))" />
-</Project>
\ No newline at end of file
+
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(ProjectLanguage)' == ''">
+    <ProjectLanguage Condition="'$(MSBuildProjectExtension)' == '.csproj' OR '$(Language)' == 'C#'">CSharp</ProjectLanguage>
+  </PropertyGroup>
+
+  <Import Project="$(MSBuildThisFileDirectory)..\Directory.Build.props" />
+  <Import Project="$(RepoToolsetDir)Settings.props" />
+
+  <PropertyGroup>
+    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
```

Comment:
It's kind of accidental that we need this everywhere now. Tasks are special because they need to copy their dependencies and apparently there's a new test requirement. We just have tasks and tests right now, but this is a scary default when we merge with CLI repo having regular class libs.

Reply:
Only two tests fail for me with CopyLocalLockFileAssemblies=false: GivenThatWeWantToBuildANetCoreAppAndPassingALogger.*

It's because the test dll passes itself as an msbuild logger,  and the msbuild loader for loggers (just like tasks) can't resolve from .deps.json. 

This could easily be fixed by moving the test logger into a tiny assembly and  having write out a simple format that doesn't need JSON.NET. e.g. "EventName=X|Prop1=Val1|...|PropN=ValN" 

I suggest:
1. Limit use of CopyLocalLockFileAssemblies to the individual test project that currently needs it.
2. File a bug to put the test logger in its own assembly and remove (1).

cc @wli3.

Reply:
Fixed.